app/lib/features/attachments/widgets/attachment_item.dart

```diff
@@ -1,41 +1,188 @@
+import 'package:acter/common/themes/app_theme.dart';
+import 'package:acter/common/utils/utils.dart';
+import 'package:acter/common/widgets/download_button.dart';
+import 'package:acter/common/widgets/image_dialog.dart';
+import 'package:acter/common/widgets/video_dialog.dart';
+import 'package:acter/features/attachments/providers/attachment_providers.dart';
+import 'package:acter/features/pins/actions/attachment_leading_icon.dart';
 import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart' show Attachment;
+import 'package:atlas_icons/atlas_icons.dart';
 import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:share_plus/share_plus.dart';
 
 // Attachment item UI
 class AttachmentItem extends ConsumerWidget {
   final Attachment attachment;
+  final bool canEdit;
+
   // whether item can be viewed on gesture
   final bool? openView;
+
   const AttachmentItem({
     super.key,
     required this.attachment,
+    this.canEdit = false,
     this.openView = true,
   });
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
-    var msgContent = attachment.msgContent();
-    String type = attachment.typeStr();
-    if (type == 'image') {
-      return AttachmentContainer(
-        name: msgContent.body(),
-        child: ImageView(attachment: attachment, openView: openView),
-      );
-    } else if (type == 'video') {
-      return AttachmentContainer(
-        name: msgContent.body(),
-        child: VideoView(attachment: attachment, openView: openView),
-      );
+    final containerColor = Theme.of(context).colorScheme.surface;
+    final borderColor = Theme.of(context).colorScheme.primary;
+    final attachmentType = AttachmentType.values.byName(attachment.typeStr());
+    final eventId = attachment.attachmentIdStr();
+    final roomId = attachment.roomIdStr();
+    final mediaState = ref.watch(attachmentMediaStateProvider(attachment));
+
+    return Container(
+      decoration: BoxDecoration(
+        color: containerColor,
+        borderRadius: BorderRadius.circular(8),
+        border: Border.all(color: borderColor),
+      ),
+      child: ListTile(
+        leading:
+            mediaState.mediaLoadingState.isLoading || mediaState.isDownloading
+                ? loadingIndication()
+                : attachmentLeadingIcon(attachmentType),
+        onTap: () => attachmentOnTap(
+          ref,
+          context,
+          attachmentType,
+          mediaState,
+        ),
+        onLongPress: canEdit
+            ? () => openRedactContentDialog(
+                  context,
+                  eventId: eventId,
+                  roomId: roomId,
+                  title: L10n.of(context).deleteAttachment,
+                  description: L10n.of(context)
+                      .areYouSureYouWantToRemoveAttachmentFromPin,
+                  isSpace: true,
+                )
+            : null,
```

Comment:
long press will delete? that is not really what we usually have, where long-press is some form of select-to-copy ...

Reply:
<img width="329" alt="Screenshot 2024-08-13 at 9 05 53 PM" src="https://github.com/user-attachments/assets/3dbc63ec-0867-4045-9ee5-097971854e39">

I have added download button indication at trailing of the item. Later stage planning to add more option menu which can be useful for Edit and Delete option so afterward I will remove delete from long press.

---
- Long-press is ideal event for addition option for any UI item. So that putted delete over there till I come-up with better attachment list UI solution.